tox.ini

```
@@ -100,6 +100,14 @@ deps =
 commands =
     python scripts/benchmark.py {posargs}
 
+[testenv:docs]
+description = Build the documentation
+deps =
+    -rdocs/requirements.txt
+commands =
+    sphinx-build -d "{toxworkdir}/docs_doctree" "{toxinidir}/docs" "{toxinidir}/docs/_output" {posargs: -j auto -bhtml}
```

How about being a bit stricter on warnings and suggesting a way to view/serve the results:
```suggestion
    # Build docs site:
    {envpython} -m sphinx \
      -d "{toxworkdir}/docs_doctree" \
      . _output \
      {posargs: -j auto -bhtml --color -a -n -W}

    # Print out the output docs dir and a way to serve html:
    {envpython} -c\
    'import pathlib;\
    docs_dir = pathlib.Path.cwd() / "_output";\
    index_file = docs_dir / "index.html";\
    print(f"\nDocumentation available under\
    `file://\{index_file\}`\n\nTo serve docs, use\
    `python3 -m http.server --directory \
    \N\{QUOTATION MARK\}\{docs_dir\}\N\{QUOTATION MARK\} 0`\n")'
changedir = docs
```

---

This seems a bit excessive to me, considering that this documentation is mostly going to live in the standard library in the end, so we won't be using this sphinx configuration for anything but the backport.

I'm somewhat on the fence about just adding `-W`. It's sometimes useful, but it's been a great source of noise in `dateutil`, because it fails whenever a link goes dead or has an HTTP 300 redirect or something.

---

I went ahead and added a "docs" GHA which incorporates a lot of the nitpickier stuff. I think most of the nitpicky stuff comes from `-b linkcheck` anyway.

Since I've removed the makefile, I'm going to keep the default `docs` environment just a thing that tries to build the docs and gives warnings rather than erroring out aggressively.